File: gemini_test/entities/item.py
from entities.post_comment import Post
from entities.brand import Brand
from typing import TypedDict

class Item(TypedDict):
    id: int
    name: str
    type: str
    description: str | None
    posts: list[Post]
    brand_id: int
    outdated: bool | None
    last_sync: str
    block_name_from_updates: bool | None

File: frontend/src/models/item.ts
import type { Brand } from "./brand"
import type { ItemAnalysisResult } from "./itemAnalysisResult"
import type { Post } from "./post"

export type ItemNoPosts = {
    id: number
    name: string
    block_name_from_updates?: boolean
    type: 'product' | 'service'
    description: string
    last_sync?: string
    posts_count?: number
    created_date: string
    updated_date: string
    item_average_score?: number
    outdated?: boolean
}

export type Item = ItemNoPosts & {
    posts?: Post[]
}

export type ItemWithPostsAndResult = Item & {
    latest_analysis_result?: ItemAnalysisResult,
    brand: Brand
    percentage_of_comments_related_to_item?: number
    negatives_count?: number
    neutral_count?: number
    positives_count?: number
}

export type ItemUpdateDto = Partial<Item>